### ISSUE TITLE
Compilation of MFEM-4.0 with CUDA support

### ISSUE THREAD

**Issue by afernandezody** [mfem/mfem#1182]
Hello,
My question is related to https://github.com/mfem/mfem/issues/1116. I couldn't find specific instructions on how to build CUDA-aware MFEM and the previous post mentions using cmake. My question is simply which is the recommended procedure to compile and build MFEM-4.0 with CUDA support.
Thank you.

**Comment by tzanio** [comment]
From https://github.com/mfem/mfem/blob/master/INSTALL:

```
- CUDA support requires an NVIDIA GPU and an installation of the CUDA Toolkit
  https://developer.nvidia.com/cuda-toolkit
```

You need to have the CUDA toolkit installed on your machine first.

CUDA-enabled build:

```
   make cuda -j 4
   (build for a specific compute capability: 'make cuda -j 4 CUDA_ARCH=sm_30')
```

Example 1 GPU run:

```
     ex1 -pa -d cuda
```

Optionally you may want to install the OCCA and RAJA libraries that will give you access to additional backends

```
- OCCA support requires the OCCA library
  https://libocca.org

- RAJA support requires installation of the RAJA performance portability layer
  with (optionally) support for CUDA and OpenMP
  https://github.com/LLNL/RAJA
```

**Comment by afernandezody** [comment]
Hi @tzanio. The latest cuda toolkit was installed, I just wanted to know about the flags. Thanks.

**Comment by sshiraiwa** [comment]
While this instruction works and runs ex1, I am having a difficulty when using CMake. What I
did is following, and the second make somehow calls g++ instead of mvcc. Am I missing something?

cmake .. -DMFEM_USE_CUDA=YES
make -j 4
make -j 4 examples

**Comment by rcarson3** [comment]
@sshiraiwa this issue related to the cmake files should be fixed in the upcoming v4.1 release as noted in a similar issue #1316 . In the mean time, you should be able to copy the modified sections of the cmake files related to the CUDA in PR #1104  and get the examples up and running.

**Comment by sshiraiwa** [comment]
I see. Thank you.
